### ISSUE TITLE
Encode screenshots for Feedback

### ISSUE THREAD

**Issue by RingerJK** [mapbox/mapbox-navigation-android#4897]
`Post Feedback API` requires an encoded screenshot. 
Need to add utils to make that for a customer who whats to provide `maps`/`entire screen` screenshots for feedback.

https://github.com/mapbox/mapbox-navigation-android/blob/31a397d6939e01109950a5149cf07f865a76cc05/libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt#L918-L937

**Comment by Zayankovsky** [comment]
I looked at how we currently capture a screenshot in 1tap and the process looks pretty sophisticated. 
At first we capture a snapshot of a map using `MapView.snapshot` function. It gives us a bitmap, which contains a map with all its objects like a puck and a route. Then we display an `ImageView` above the map and load the snapshot into it. And finally we capture a screenshot using any standard means, for example via `View.getDrawingCache`. This lets us have a screenshot which contains not only map, but also all views displayed on top of it, like banner instructions and trip progress. 
So to implement this we need to place an `ImageView` right about the `MapView`, for example like this:
```xml
<com.mapbox.maps.MapView
    android:id="@+id/mapView"
    android:layout_width="0dp"
    android:layout_height="0dp"
    app:layout_constraintBottom_toBottomOf="parent"
    app:layout_constraintEnd_toEndOf="parent"
    app:layout_constraintStart_toStartOf="parent"
    app:layout_constraintTop_toTopOf="parent" />

<ImageView
    android:id="@+id/snapshotView"
    android:layout_width="0dp"
    android:layout_height="0dp"
    app:layout_constraintBottom_toBottomOf="@id/mapView"
    app:layout_constraintEnd_toEndOf="@id/mapView"
    app:layout_constraintStart_toStartOf="@id/mapView"
    app:layout_constraintTop_toTopOf="@id/mapView" />
```
Then in order to capture a screenshot we do something like this:
```kotlin
mapView.snapshot { snapshot ->
    if (snapshot != null) {
        runOnUiThread {
            snapshotView.setImageBitmap(snapshot)
            snapshotView.isVisible = true
            val rootView = mapView.rootView
            rootView.isDrawingCacheEnabled = true
            val screenshot = Bitmap.createBitmap(root.drawingCache)
            rootView.isDrawingCacheEnabled = false
            snapshotView.isVisible = false
            snapshotView.setImageDrawable(null)
        }
    }
}
```
Two attached images below showcase the difference between what `MapView.snapshot` function provides (left) and the final result (right). 
![1633523896860-snapshot](https://user-images.githubusercontent.com/2395284/136203768-aa46dae7-b786-4104-a527-697817046ad1.png) ![1633523897053-capture](https://user-images.githubusercontent.com/2395284/136203770-8d49861f-9f25-4be0-a548-479184f83795.png)
cc @RingerJK @Guardiola31337

**Comment by Guardiola31337** [comment]
That's correct. Commented here https://github.com/mapbox/mapbox-navigation-android/pull/4942#discussion_r723201410 `snapshot` only gets a screenshot of the map portion, so in order to get a full screenshot including the rest of the components we need to implement that workaround 🚀
